overcooked_ai_py/agents/benchmarking.py

```diff
@@ -20,46 +20,71 @@ class AgentEvaluator(object):
     pickleable. We should think about possible improvements/what makes most sense to do here.
     """
 
-    def __init__(self, mdp_params, env_params={}, mdp_fn_params=None, force_compute=False, mlp_params=NO_COUNTERS_PARAMS, debug=False):
+    def __init__(self, mdp_params_lst, env_params={}, mdp_fn_params_lst=None, force_compute=False, mlp_params_lst=None, debug=False):
         """
-        mdp_params (dict): params for creation of an OvercookedGridworld instance through the `from_layout_name` method
+        mdp_params_lst (list): a list of params for creation of an OvercookedGridworld instance
+            through the `from_layout_name` method
         env_params (dict): params for creation of an OvercookedEnv
-        mdp_fn_params (dict): params to setup random MDP generation
+        mdp_fn_params (list): a list of params to setup random MDP generation
         force_compute (bool): whether should re-compute MediumLevelPlanner although matching file is found
         mlp_params (dict): params for MediumLevelPlanner
         """
-        assert type(mdp_params) is dict, "mdp_params must be a dictionary"
-
-        if mdp_fn_params is None:
-            mdp = OvercookedGridworld.from_layout_name(**mdp_params)
-            self.mdp_fn = lambda: mdp
-            self.env = OvercookedEnv.from_mdp(mdp, **env_params)
+        assert type(mdp_params_lst) is list and len(mdp_params_lst) > 0, \
+            "please make sure mdp_params_lst is a list. " \
+            "If not, please see where the function is called and make a fake list"
+        assert all([type(mdp_params_i) is dict for mdp_params_i in mdp_params_lst]), \
+            "all mdp_params must be a dictionary"
+
+        # The main changes here is that we are putting a list of environment in the AgentEvaluator.
+        self.mdp_fn_lst = []
+        self.env_lst = []
+        if mdp_fn_params_lst is None:
+            for mdp_params_i in mdp_params_lst:
+                mdp = OvercookedGridworld.from_layout_name(**mdp_params_i)
+                self.mdp_fn_lst.append(lambda: mdp)
+                self.env_lst.append(OvercookedEnv.from_mdp(mdp, **env_params))
         else:
-            self.mdp_fn = LayoutGenerator.mdp_gen_fn_from_dict(mdp_params, **mdp_fn_params)
-            self.env = OvercookedEnv(self.mdp_fn, **env_params)
+            assert type(mdp_fn_params_lst) is list and len(mdp_fn_params_lst) > 0
+            assert all([type(mdp_fn_params_i) is dict for mdp_fn_params_i in mdp_fn_params_lst]), \
+                "all mdp_fn_params must be a dictionary"
+            for mdp_params_i, mdp_fn_params_i in zip(mdp_params_lst, mdp_fn_params_lst):
+                mdp_fn = LayoutGenerator.mdp_gen_fn_from_dict(mdp_params_i, **mdp_fn_params_i)
+                self.mdp_fn_lst.append(mdp_fn)
+                self.env_lst.append(OvercookedEnv(mdp_fn, **env_params))
```

Comment:
Eventually here you are basically doing the same thing as was done before (using LayoutGenerator for each mdp_params dict), but all this functionality could be embedded in the LayoutGenerator and the mdp_fn_params I believe?

Reply:
Agreed. I noticed the similarity, but I just strictly followed the original code for this version. I will make an update to it after I fix the LayoutGenerator.